go/vt/vtadmin/api.go

```diff
@@ -956,6 +958,45 @@ func (api *API) GetTablet(ctx context.Context, req *vtadminpb.GetTabletRequest)
 	return nil, vterrors.Errorf(vtrpcpb.Code_NOT_FOUND, "%s: %s, searched clusters = %v", errors.ErrAmbiguousTablet, req.Alias, ids)
 }
 
+// PingTablet is part of the vtadminpb.VTAdminServer interface.
+func (api *API) PingTablet(ctx context.Context, req *vtadminpb.PingTabletRequest) (*vtadmin.PingTabletResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.PingTablet")
+	defer span.Finish()
+
+	tablet, err := api.GetTablet(ctx, &vtadminpb.GetTabletRequest{
```

Comment:
This reuse is really smart but it's exposing something I didn't think through with rbac: if a user has permission on whatever resource (see my comment below) allows Ping, but does not have the GetTablet permission, we will fail here. Maybe that's fine? We can also include in the documentation (ya know, when we write down the various permissions at some point) that in order to PingTablet you also need explicitly to be able to GetTablet, or maybe we should refactor `GetTablet` to allow this reuse without needing it, which could look like:

```go
func (api *API) GetTablet(...) (*GetTabletResponse, error) {
    if !api.authz.IsAuthorized(..., TabletResource, GetAction) {
        fail
    }

    return api.getTablet(ctx, req)
}

func (api *API) getTablet(ctx, req) (*GetTabletResponse, error) {
    // rest of original GetTablet logic goes here
}
```

and then call just the `getTablet` which would skip the extra authz check

Reply:
Actually this might take more refactoring since the auth check happens during the loop over clusters...

Reply:
yeah .. after thinking about this some more it's probably not just Fine for it to be this way but actually better ... because if you could Ping without Get then I could in theory use the Ping apis to enumerate tablets even in clusters where i'm not allowed to Get them